Source/WebCore/rendering/style/BasicShapes.h

```diff
@@ -385,6 +387,57 @@ class BasicShapeInset final : public BasicShape {
     LengthSize m_bottomLeftRadius;
 };
 
+class BasicShapeRect final : public BasicShape {
+public:
+    static Ref<BasicShapeRect> create() { return adoptRef(*new BasicShapeRect); }
+    WEBCORE_EXPORT static Ref<BasicShapeRect> create(Length&& top, Length&& right, Length&& bottom, Length&& left, LengthSize&& topLeftRadius, LengthSize&& topRightRadius, LengthSize&& bottomRightRadius, LengthSize&& bottomLeftRadius);
+
+    Ref<BasicShape> clone() const final;
+
+    const Length& top() const { return m_edges.top(); }
+    const Length& right() const { return m_edges.right(); }
+    const Length& bottom() const { return m_edges.bottom(); }
+    const Length& left() const { return m_edges.left(); }
+
+    const LengthSize& topLeftRadius() const { return m_topLeftRadius; }
+    const LengthSize& topRightRadius() const { return m_topRightRadius; }
+    const LengthSize& bottomRightRadius() const { return m_bottomRightRadius; }
+    const LengthSize& bottomLeftRadius() const { return m_bottomLeftRadius; }
+
+    void setTop(Length top) { m_edges.setTop(WTFMove(top)); }
+    void setRight(Length right) { m_edges.setRight(WTFMove(right)); }
+    void setBottom(Length bottom) { m_edges.setBottom(WTFMove(bottom)); }
+    void setLeft(Length left) { m_edges.setLeft(WTFMove(left)); }
+
+    void setTopLeftRadius(LengthSize radius) { m_topLeftRadius = WTFMove(radius); }
+    void setTopRightRadius(LengthSize radius) { m_topRightRadius = WTFMove(radius); }
+    void setBottomRightRadius(LengthSize radius) { m_bottomRightRadius = WTFMove(radius); }
+    void setBottomLeftRadius(LengthSize radius) { m_bottomLeftRadius = WTFMove(radius); }
+
+private:
+    BasicShapeRect() = default;
+    BasicShapeRect(RectEdges<Length>&&, LengthSize&& topLeftRadius, LengthSize&& topRightRadius, LengthSize&& bottomRightRadius, LengthSize&& bottomLeftRadius);
+    BasicShapeRect(Length&& top, Length&& right, Length&& bottom, Length&& left, LengthSize&& topLeftRadius, LengthSize&& topRightRadius, LengthSize&& bottomRightRadius, LengthSize&& bottomLeftRadius);
+
+    Type type() const override { return Type::Rect; }
+
+    const Path& path(const FloatRect&) override;
+
+    bool canBlend(const BasicShape&) const override;
+    Ref<BasicShape> blend(const BasicShape& from, const BlendingContext&) const override;
+
+    bool operator==(const BasicShape&) const override;
+
+    void dump(TextStream&) const final;
```

Comment:
The class is final, so you may as well make all these say final instead of override.